content/posts/2022-11-09-iOS16-lock-screen-dynamic-time/index.md

```diff
@@ -0,0 +1,185 @@
+
+Let’s now fix it. We don’t want a broken clock in our application. Do we?
+
+## Update the date-time
+
+We need a couple of changes now in order to update the time.
+
+### Create a date-time state
+
+1. Since we will need to update our time, we need to put it into a state variable.
+
+```jsx
+import { useState } from 'react';
+
+// inside our component
+const [date, setDate] = useState(dayjs()); // default value is now our current time
+```
+
+### Format our date-time using dayjs
+
+2. We have put `dayjs()` into our `date` state so we can use our `date` state to show the date-time now inside our `<Text />` component
+
+```jsx
+<Text style={styles.date}>{date.format("dddd, DD MMMM")}</Text>
+<Text style={styles.time}>{date.format("hh:mm")}</Text>
+
+```
+
+### Update the date-time state using `setInterval`
+
+3. We want to update the time with one minute interval every time. To do so we can use the javascript’s `setInterval` function. A typical `setInterval` function’s prototype is the following,
+
+```jsx
+setInterval(() => {
+  functionToTrigger();
+}, timeInMs);
+```
+
+This means that the `functionToTrigger()` will be called or triggered every `timeInMs` value. We will set the value of `timeInMs` and the `functionToTrigger` now.
+
+As we want to update our clock every minute, our function will become,
+
+```jsx
+setInterval(() => {
+  updateTime(); // dummy function for now, will work on this.
+}, 60000); // 1 min = 60000 milliseconds
+```
+
+Great, now our `updateTime` function will trigger an update in every one minute. Let’s update our time by using `setDate` hook.
+
+```jsx
+setInterval(() => {
+  setDate(dayjs()); // dayjs() always gives us the current time
+}, 60000);
+```
+
+4. We have our core logic set up for updating the time, where do we want to use this logic now? You might have guessed it already, since this is a side effect and it does not depend on user’s interaction we will use react’s `useEffect` hook. The next question is when do we want to start this logic? We want to start our timer after our component mounts. By using an empty dependency array inside our `useEffect` hook we can achieve that lifecycle.
+
+```jsx
+useEffect(() => {
+  setInterval(() => {
+    setDate(dayjs()); // dayjs() always gives us the current time
+  }, 60000);
+}, []);
+```
+
+Awesome, with that you should be able to see your time being updated every minute. Exciting!
+
+### Preview
+
+Before we go any further, let’s check whether our implementation is working so far, let’s update the time in every seconds and show seconds in our header with the hours and mins for the sake of the demo.
+
+```jsx
+useEffect(() => {
+  setInterval(() => {
+    setDate(dayjs()); // dayjs() always gives us the current time
+  }, 1000 * 1);
```

Comment:
Here you can remove the comment // dayjs()..., and add a comment in the last line to emphasize what should be changed.